crates/bevy_ecs/src/system/commands/command_queue.rs

```diff
@@ -81,27 +85,11 @@ impl CommandQueue {
         // unnecessary allocations.
         unsafe { self.bytes.set_len(0) };
 
-        let byte_ptr = if self.bytes.as_mut_ptr().is_null() {
```

Comment:
Going by what the docs say I don't think we can program against them, `as_mut_ptr` could also return a freed pointer, which according to https://doc.rust-lang.org/stable/std/ptr/index.html#safety are also not valid for zero-sized reads. I guess the correct way to do this would be checking if the capacity is 0 rather than if the pointer is null. Anyway, I opened a PR to change `Vec::as_ptr` and `Vec::as_mut_ptr`'s docs, so hopefully this get clarified in the stdlib.